test/common/upstream/health_checker_impl_test.cc

```diff
@@ -4414,6 +4426,21 @@ TEST_F(GrpcHealthCheckerImplTest, GrpcFailUnknown) {
             cluster_->prioritySet().getMockHostSet(0)->hosts_[0]->health());
 }
 
+// This used to cause a null dereference
+TEST_F(GrpcHealthCheckerImplTest, GrpcFailNullDereference) {
```

Comment:
nit: could you make the test name descriptive of the problem? like `GrpcFailResponseNullBytes`?

Reply:
Changed to GrpcFailResponseNullBytes